Sources/SecureXPC/Server/XPCServer.swift

```diff
@@ -128,6 +133,13 @@ public class XPCServer {
             connections.compactMap{ $0.connection }.forEach{ xpc_connection_set_target_queue($0, newValue) }
         }
     }
+    // MARK: Route registration
+        
+    private func checkUniqueRoute(route: XPCRoute) throws {
+        if self.routes.keys.contains(route) {
```

Comment:
IDK how much it matters, but these two operations (checking it's not already there, and setting the new value) could be done in a single hash look-up with https://developer.apple.com/documentation/swift/dictionary/3127179-updatevalue

Reply:
Eh yeah I don't think any efficiency gains matter here, I'd rather consolidate the error generation into just one place.